src/tcpserver.c

```diff
@@ -161,11 +154,14 @@ static void tcplistener_accept_callback(struct ev_loop *loop,
 
 	session->ctx = listener->cb_conn(session->sd, session->data);
 
+	// Save the new connection socket descriptor
+	// into the vector, for future cleanup!
+	vector_insert(&listener->sdList, (void *)session);
```

Comment:
Simplification: file descriptors are returned lowest-integer-first always. We can index directly into our vector by the resulting file descriptor, and don't need to do linear searches.

i.e.:

`vector_set(vector, fd, (void*)watcher);`

where the data for unset vector elements would by NULL.